housekeeper/store/api/find.py

```diff
@@ -76,12 +77,25 @@ def version(
             bundle_name=bundle,
         ).first()
 
-    def tag(self, name: str) -> Tag:
-        """Fetch a tag from the database."""
-        return self.Tag.filter_by(name=name).first()
+    def tag(self, tag_name: str = None) -> Tag:
+        """Return a tag from the database."""
+        if tag_name:
+            LOG.info(f"Fetching tag with name: {tag_name}")
+            return apply_tag_filter(
+                tags=self._get_tag_query(),
+                filter_functions=[TagFilters.FILTER_BY_NAME],
+                tag_name=tag_name,
+            ).first()
+
+        return self._get_tag_query().first()
```

Comment:
Do we really want to return a random tag when the tag name is None?

Reply:
Good question! We could consider changing that functionality

Reply:
I think it does not make sense and the original function does not behave like this. So you can just remove this part and the if statement